File: exam_at_home/1/swing.cc
/*
* author: wang-yang
* email: tnst92002@gmail.com
*/
#include <bits/stdc++.h>
using namespace std;

vector<int> run(vector<int> s) {
    int cnt[26] = {0};
    vector<int> res;
    for (int c : s)
        cnt[c]++;
    bool Suc;
    int order = 0;
    do {
        Suc = false;
        if (order == 0) {
            for (int i=0;i<26;++i) {
                if (cnt[i]) {
                    Suc = true;
                    cnt[i]--;
                    res.push_back(i);
                }
            }
        } else {
            for (int i=25;i>=0;--i) {
                if (cnt[i]) {
                    Suc = true;
                    cnt[i]--;
                    res.push_back(i);
                }
            }
        }
        order = 1 - order;
    } while (Suc);
    return res;
}

int main() {
    int N;
    cin >> N;
    while (N--) {
        int len;
        cin >> len;
        vector<int> buf(len);
        for (int &i: buf) cin >> i;
        auto res = run(buf);
        for (size_t i=0; i<res.size();++i)
        {
            if (i) cout << ' ';
            cout << res[i];
        }
        cout << '\n';
    }
}

File: exam_at_home/1/swing.py
#
# author: wang-yang
# email: tnst92002@gmail.com
#
N = int(input())

def run(s: str):
    res = []
    cnt = [0] * 26
    for c in s:
        cnt[c] += 1
    order = 0
    while True:
        end = True
        if order == 0 :
            for i in range(26):
                if cnt[i] != 0:
                    end = False
                    cnt[i] = cnt[i] - 1
                    res.append(i)
        else:
            for i in range(25, -1, -1):
                if cnt[i] != 0:
                    end = False
                    cnt[i] = cnt[i] - 1
                    res.append(i)
        order = 1 - order
        if end:
            break
    return res

for _ in range(N):
    len, *arr = map(int, input().split())
    res = run(arr)
    print(' '.join(map(str,res)))